flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeJsonDeserializer.java

```diff
@@ -506,6 +529,29 @@ private static boolean isTemporary(SqlOperator sqlOperator) {
         }
     }
 
+    private static Optional<SqlOperator> lookupOptionalSqlOperator(
+            String operatorName, SqlSyntax syntax, @Nullable SqlKind sqlKind) {
+        List<SqlOperator> foundOperators = new ArrayList<>();
+        // try to find operator from std operator table.
+        SqlStdOperatorTable.instance()
+                .lookupOperatorOverloads(
+                        new SqlIdentifier(operatorName, new SqlParserPos(0, 0)),
+                        null, // category
+                        syntax,
+                        foundOperators,
+                        SqlNameMatchers.liberal());
+        if (foundOperators.size() == 1) {
```

Comment:
Could this similify to `foundOperators.stream().filter(operator->sqlKind != null && operator.kind == sqlKind).findFirst()`?